mlflow/pyfunc/scoring_server/client.py

```diff
@@ -55,3 +56,12 @@ def invoke(self, data):
             )
 
         return scoring_server.infer_and_parse_json_input(response.text)
+
+    def _predict(self, data):
+        """
+        Invoke inference on input data and return the predicted column as a pandas series
+        """
```

Comment:
Made this a private function for now given it is only used internally. @dbczumar let me know if this comment is accurate. It seems like the client invocation will result in a pandas dataframe in most cases. So in most cases, this function would return a pandas series.

Reply:
I do wonder if there are cases that a model returns an ArrayType, similar to what was shown here: https://src.dev.databricks.com/mlflow/mlflow@a0f43ff172206ea1432179fbbe7ae1b9a8df639e/-/blob/mlflow/pyfunc/__init__.py?L925-928

Reply:
Can we place this logic in `_ServedPyFuncModel.predict()`?

Reply:
In this case, would we override the [schema enforcement logic](https://src.dev.databricks.com/mlflow/mlflow@a0f43ff172206ea1432179fbbe7ae1b9a8df639e/-/blob/mlflow/pyfunc/__init__.py?L364-381) of the PyFuncModel predict function?